backend/src/main/java/gov/cdc/usds/simplereport/service/email/SendGridWrapper.java

```diff
@@ -0,0 +1,49 @@
+package gov.cdc.usds.simplereport.service.email;
+
+import com.sendgrid.Method;
+import com.sendgrid.Request;
+import com.sendgrid.Response;
+import com.sendgrid.SendGrid;
+import com.sendgrid.helpers.mail.Mail;
+import java.io.IOException;
+import javax.annotation.PostConstruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Component;
+
+@ConditionalOnProperty(name = "sendgrid.enabled", havingValue = "true")
+@Primary
+@Component
+public class SendGridWrapper implements EmailProviderWrapper {
```

Comment:
It seems like it'd be simpler if you just had this method check `sendgrid.enabled` directly, and return null (or raise an exception) as required.

Reply:
Simpler, sure, but isn't it more Spring-like to use configuration values (or even profiles) to inject differently-behaving components of the same interface for different environments?